docs/deployment/pull-backup.rst

```diff
@@ -301,3 +301,101 @@ a backup may be the following command::
 
 This command also automatically removes the socket file after the ``borg
 create`` command is done.
+
+ssh-agent
+=========
+
+In this scenario *borg-server* initiate SSH connection to *borg-client* with forwarding of the authentication agent connection.
+Afterwards scenario is similar to the push mode: *borg-client* initiate another SSH connection
+back to *borg-server* using forwarded agent connection for authenticate itself,
+starts ``borg serve`` and communicate with them.
+
+Using of this method requires ssh access of user *borgs* to *borgc@borg-client*. Where:
+
+* *borgs* is the user on the server side with read/write access to local borg repository.
+* *borgc* is the user on the client side with read access to files meant to be backed up.
+
+Apply of this method in case of automated backup operations
+-----------------------------------------------------------
+
+Assume that borg-client host is untrusted.
+Therefore there is effort to prevent hostile user on borg-client side to do something harmful.
+In case of fully trusted borg-client the method may be simplified.
+
+Prepare server side
+~~~~~~~~~~~~~~~~~~~
+
+Do this once for each client on *borg-server* for allowing *borgs* to connect itself on *borg-server* using dedicated RSA key::
+
+  borgs@borg-server$ install -m 700 -d ~/.ssh/
+  borgs@borg-server$ ssh-keygen -N '' -t rsa  -f ~/.ssh/borg-client_rsa
+  borgs@borg-server$ { echo -n 'command="borg serve --append-only --restrict-to-path ~/repo",restrict '; cat ~/.ssh/borg-client_rsa.pub; } >> ~/.ssh/authorized_keys
+  borgs@borg-server$ chmod go-w ~/.ssh/authorized_keys
+
+``install -m 700 -d ~/.ssh/``
+
+  Create directory ~/.ssh with correct permissions if not exists yet.
+
+``ssh-keygen -N '' -t rsa  -f ~/.ssh/borg-client_rsa``
+
+  Create RSA key dedicated to communication with borg-client.
+
+``{ echo -n 'command="borg serve --append-only --restrict-to-path ~/repo",restrict '; cat ~/.ssh/borg-client_rsa.pub; } >> ~/.ssh/authorized_keys``
```

Comment:
Yes, it works in my case, sshd executes command in user's shell and expansion works. I agree, this may be not reliable method as it depends on shell expansion behavior, which may not be bash in common case. But I already made implicit assumption that borgs's shell is bash when using syntax like '~/.ssh/borg-client_rsa' or parentheses around commands.

Looks like sshd just concatenate with space all parameters passed as commands to ssh and starts <user shell> -c '<concatenated commands>'.

I can replace ~ with absolute paths or keep it as is for compliance with ssh://borgs@borg-server/~/repo syntax.

Reply:
if it works, keep it as is. was just unsure...